### PR TITLE
Feat/alert location clearer

### PR DESCRIPTION
**Pull request by j8seangel**
New map layout with the following changes:
- Map performance improved to avoid unnecessary re-renders
- Center to position icon to fit to position or position and alert when having at least one selected
- Removed the rarely used areas carousel moving the area settings to the sidebar
- New sidebar with area options and legend

Also:
- Not crashing when no token for logout.
- Fix ios build updating FastImage dependency.
- Update broken reactotron dependency for me (please test yours).

image
image

### CODE REVIEW

File: app/components/common/alert-legend/index.js [hunk 1]
```diff
@@ -0,0 +1,25 @@
+import React from 'react';
+import PropTypes from 'prop-types';
```
Comment (sorodrigo) [inline app/components/common/alert-legend/index.js]: lets use flow instead of proptypes

File: app/components/map/index.js [hunk 2]
```diff
@@ -148,12 +144,11 @@ class Map extends Component {
       !isEqual(nextProps.areaCoordinates, this.props.areaCoordinates),
       !isEqual(nextProps.area, this.props.area),
       nextProps.canDisplayAlerts !== this.props.canDisplayAlerts,
-      !isEqual(nextProps.center, this.props.center),
       !isEqual(nextProps.contextualLayer, this.props.contextualLayer),
       !isEqual(nextState.lastPosition, this.state.lastPosition),
       nextState.hasCompass !== this.state.hasCompass,
       nextState.heading !== this.state.heading,
-      !isEqual(nextState.region, this.state.region),
+      // !isEqual(nextState.region, this.state.region),
```
Comment (sorodrigo) [inline app/components/map/index.js]: if you don't use it, you lose it. 😛

File: app/components/map/index.js [hunk 3]
```diff
@@ -163,15 +158,17 @@ class Map extends Component {
...
-    const { clusters } = clusterGenerator;
-    if (this.map && area && area.dataset && (clusters === null || area.id !== prevProps.area.id
-      || !isEqual(area.dataset, prevProps.area.dataset))) {
-      setActiveAlerts();
-    }
-
-    if (!isEqual(areaCoordinates, prevProps.areaCoordinates)) {
-      this.updateSelectedArea();
+    const { area, setActiveAlerts } = this.props;
+    if (area && area.id && area.dataset) {
```
Comment (sorodrigo) [inline app/components/map/index.js]: do we need to check if `area.id` exists?

File: app/components/map/index.js [hunk 4]
```diff
@@ -212,13 +208,13 @@ class Map extends Component {
         this.onDidAppear();
         break;
       case 'contextualLayers':
```
Comment (sorodrigo) [inline app/components/map/index.js]: no longer used, can be removed.

File: app/components/map/index.js [hunk 5]
```diff
@@ -438,10 +444,11 @@ class Map extends Component {
   }
 
   updateRegion = (region) => {
-    const mapZoom = Map.getMapZoom(region);
-    const clean = this.state.mapZoom > mapZoom;
+    const mapZoom = MapComponent.getMapZoom(region);
+    // const clean = this.state.mapZoom > mapZoom;
```
Comment (sorodrigo) [inline app/components/map/index.js]: So glad this is gone. Let's delete this line too!

File: app/components/map/index.js [hunk 6]
```diff
@@ -514,51 +521,85 @@ class Map extends Component {
...
+            kmThreshold={30}
           />
-        );
-    };
+        </View>
+        <View style={styles.buttonPanelRow}>
+          <View pointerEvents="box-none" style={styles.btnContainer}>
+            {neighbours && neighbours.length > 0
+              ? [
+                <CircleButton
```
Comment (sorodrigo) [inline app/components/map/index.js]: Prefer fragment

File: app/components/map/index.js [hunk 7]
```diff
@@ -574,17 +615,38 @@ class Map extends Component {
...
 
+  renderMapFooter() {
+    const { selectedAlerts, neighbours } = this.state;
+    const hasAlertsSelected = selectedAlerts && selectedAlerts.length > 0;
+
+    const hasNeighbours = neighbours && neighbours.length > 0;
+    let veilHeight = 120;
+    if (hasAlertsSelected) veilHeight = hasNeighbours ? 260 : 180;
+
+    return [
```
Comment (sorodrigo) [inline app/components/map/index.js]: prefer fragment

File: app/components/map/index.js [hunk 8]
```diff
@@ -730,10 +793,11 @@ class Map extends Component {
           minZoomLevel={2}
           maxZoomLevel={18}
           initialRegion={region}
-          rotateEnabled={false}
-          onRegionChangeComplete={this.updateRegion}
-          onMapReady={this.onMapReady}
+          showsCompass
```
Comment (sorodrigo) [inline app/components/map/index.js]: Are we using the compass?

File: app/components/map/map-attribution/styles.js [hunk 9]
```diff
@@ -6,16 +6,21 @@ import {
 export default StyleSheet.create({
   attributionContainer: {
     position: 'absolute',
+    left: 0,
     right: 0,
-    bottom: 4,
```
Comment (sorodrigo) [inline app/components/map/map-attribution/styles.js]: this hides the google attribution, I think we still need it.

File: app/components/map/index.js (hunk 8 above)
Comment (j8seangel) [reply app/components/map/index.js]: This is the top right circle that will appear on the map when you rotate it and it helps you to know where the north is.

File: app/components/map/index.js (hunk 2 above)
Comment (j8seangel) [reply app/components/map/index.js]: But we aren't using it on the map so it is working perfectly and we are avoiding re-renders

Comment (sorodrigo) [reply app/components/map/index.js]: I mean we can remove the commented line

File: app/components/map/map-attribution/styles.js (hunk 9 above)
Comment (sorodrigo) [reply app/components/map/map-attribution/styles.js]: looks like it's no longer needed 👍

**Pull request merged**